### PR TITLE
kafka: dependency upgrade to 3.1.0

### PR DESCRIPTION
**Pull request by adamkotwasinski**
Commit Message: kafka: dependency upgrade to 3.1.0
Additional Description: Upgrade the Kafka dependencies we use to generate protocol-parsing code 3.0.0 -> 3.1.0
Risk Level: low
Testing: integration (with new server); manual (server = 3.1.0 + client = 1.1.1 / 3.1.0)
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by phlax** [comment]
im wondering why this is not being picked up as an out-of-date dep - defo not a blocker to this PR tho

**Comment by htuch** [comment]
@adamkotwasinski is the dep upgrade related to the code changes or are they independent? If the latter, prefer separate PRs.

File: contrib/kafka/filters/network/source/protocol/generator.py [hunk 1]
```diff
@@ -534,7 +534,7 @@ class Primitive(TypeSpecification):
     # Custom values that make test code more readable.
     KAFKA_TYPE_TO_EXAMPLE_VALUE_FOR_TEST = {
         'string':
-            '"string"',
+            'static_cast<std::string>("string")',
```
Comment (adamkotwasinski) [inline contrib/kafka/filters/network/source/protocol/generator.py:537]: FWIW This is necessary for test code (example values) to compile (as with changing request declarations we ran into a situation where constructor invocations are ambiguous in https://github.com/envoyproxy/envoy/blob/v1.21.0/contrib/kafka/filters/network/test/protocol/response_utilities_cc.j2#L43).

**Comment by adamkotwasinski** [comment]
ready for review

**Review by htuch — APPROVED** [review]
/lgtm deps

**Pull request merged**